### PR TITLE
Rad badge fix

### PR DESCRIPTION
**Pull request by bkentel**
- `rad_threshold_colors` was defined needlessly in multiple TUs (see initial commit).
- existing code didn't allow jumps in color change of more than one.
- existing code didn't check for repeated messages.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -1246,15 +1269,9 @@ std::string item::info(bool showtext, std::vector<iteminfo> *dump, bool debug) c
...
-            size_t i;
-            for( i = 1; i < sizeof(rad_dosage_thresholds) / sizeof(rad_dosage_thresholds[0]); i++ ) {
-                if( irridation < rad_dosage_thresholds[i] ) {
-                    break;
-                }
-            }
             dump->push_back(iteminfo("DESCRIPTION",
                 string_format(_("The film strip on the badge is %s."),
-                              _(rad_threshold_colors[i - 1].c_str()))));
+                              _(rad_badge_color(irridation).c_str()))));
```
Comment (BevapDin) [inline src/item.cpp]: Looks like a double translations (same below), the result of `rad_badge_color` is already translated as the `values` array contains translated entries, doesn't it?

Btw. a C-style array? Why not a C++ array?

Comment (bkentel) [inline src/item.cpp]: Yep, double translation; good eye! I could use a std::array here, but range for works with C arrays too.

Comment (kevingranade) [inline src/item.cpp]: Need to go the other way with this (leave initialized values translated and avoid translating here), as it is now the gettext system is not necessarily going to be able to extract the strings from the code.  In reality I'm pretty sure all the colors are covered elsewhere, but it's still bad form.

Comment (bkentel) [inline src/item.cpp]: Sure, I'll flip it around. I'm not terribly familiar with how gettext works, so I didn't give it a lot of consideration.

**Comment by bkentel** [comment]
Flipped the translation location.

**Comment by kevingranade** [comment]
Stellar, thanks.

**Pull request merged**